packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -462,6 +517,25 @@ function specialCaseRefs(value: any): any {
   }
 }
 
+function specialCaseSubRefs(value: any): any {
+  switch (value) {
+    case 'AWS::AccountId': return pseudoString(value);
+    case 'AWS::Region': return pseudoString(value);
+    case 'AWS::Partition': return pseudoString(value);
+    case 'AWS::URLSuffix': return pseudoString(value);
+    case 'AWS::NotificationARNs': return pseudoString(value);
+    case 'AWS::StackId': return pseudoString(value);
+    case 'AWS::StackName': return pseudoString(value);
+    case 'AWS::NoValue': return pseudoString(value);
+    case 'AWS::NoValue': return pseudoString(value);
+    default: return undefined;
+  }
+}
+
+function pseudoString(value: any): any {
+  return Token.asString('${' + value + '}', {displayHint: value });
+}
+
```

Comment:
I see that your suggested change works, but I don't understand why.

The constants in `specialCaseRefs()` are tokens, so why are these different? Don't these pseudo parameters still need to be resolved as they are in `specialCaseRefs()`?

Reply:
They are different, because a customer can use `Aws.REGION` when defining their resource, like this:

```ts
myBucket.bucketName = Aws.REGION + "abcd";
```

If `Aws.REGION` was just a JS object that resolves to `{ Ref: 'AWS::Region' }`, obviously that code would not compile (maybe it would even compile, but it would not do the right thing an synth time for sure).

That is not a concern when parsing the string provided to `Fn::Sub` though 🙂.

Does this explanation make sense?

Reply:
It helps, but I'm wondering; does this mean that we don't have to turn pseudo parameters in `Fn::Sub` expressions into tokens at all?

Reply:
You don't have to turn them into tokens for the same reason you don't have to turn references to the map-defined keys (so `"Fn::Sub": ["${a}", { a: "sth" }]`) into tokens.

A token is a late-bound value (that is, one that will be resolved at synth time). Since you know `${a}` above will always resolve to the string `"${a}"` (and same for `${AWS::Region}`), a token does not add anything in those cases - it's superfluous.

But it's a different story with `Aws.REGION` (which is a string at compile time, but resolves to `{ Ref: 'AWS::Region' }` at synth time).

Reply:
ohh ok. Thanks, that makes sense.

Reply:
To be clear - references to Resources and Parameters (so, `{ "Fn::Sub": "${Bucket} ${Bucket.DomainName}" }`), though, absolutely do need to be tokens, because the logical ID can, in fact, change at synth time.